File: CodeForces/1535b/main.cpp
#include <bits/stdc++.h>
#include <iostream>
#include <vector>

using namespace std;

int gcd(int a, int b) {
    if(a == 0) {
        return b;
    }
    return gcd(b % a, a);
}

void solve() {

    int n, ans = 0;
    vector<int> arr;
    vector<int> temp;

    cin >> n;

    for(int i = 0; i < n; i++) {
        int x;
        cin >> x;
        if(x % 2 == 0) {
            arr.emplace_back(x);
        } else {
            temp.emplace_back(x);
        }
    }

    for(int i = 0; i < temp.size(); i++) {
        arr.emplace_back(temp[i]);
    }

    for(int i = 0; i < n; i++) {
        for(int j = i+1; j < n; j++) {
            if(gcd(arr[i], 2 * arr[j]) > 1) {
                ans++;
            }
        }
    }

    cout << ans << "\n";

}

int main() {
    int t;
    cin >> t;
    while(t--) {
        solve();
    }
}

File: CodeForces/1535b/main.py
def inp():
    return int(input())


def inlt():
    return list(map(int, input().split()))


def gcd(a, b):
    if a == 0:
        return b
    return gcd(b % a, a)


def solve():
    n = inp()
    arr = inlt()
    t1, t2 = [], []
    ans = 0

    for x in arr:
        if x % 2 == 0:
            t1.append(x)
        else:
            t2.append(x)

    arr = t1 + t2

    for i in range(n):
        for j in range(i + 1, n):
            if gcd(arr[i], 2 * arr[j]) > 1:
                ans += 1

    print(ans)

    return


t = inp()

for _ in range(t):
    solve()
